server/src/test/java/org/elasticsearch/action/search/SearchAsyncActionTests.java

```diff
@@ -142,15 +142,21 @@ protected SearchPhase getNextPhase(SearchPhaseResults<TestSearchPhaseResult> res
                     return new SearchPhase("test") {
                         @Override
                         public void run() {
-                            latch.countDown();
```

Comment:
If we remove this then how do we know the search phase actually runs?

Reply:
yes we are just moving to counting the search the shards rather than the one search, which was counted too early.

Reply:
How about making an `AtomicBoolean searchPhaseDidRun` and then calling `assertTrue(searchPhaseDidRun.compareAndSet(false, true));` here and `assertTrue(searchPhaseDidRun.get());` somewhere after the `latch.await();`?

Reply:
yes that makes sense. I initially did not consider that we were using the latch to also make sure that we would move on to the next phase. It makes sense to check that.